selftests/functional/test_test.py

```diff
@@ -0,0 +1,69 @@
+import json
+import os
+import shutil
+import tempfile
+import unittest
+
+from avocado.core import exit_codes
+from avocado.utils import process
+from avocado.utils import script
+
+basedir = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..')
+basedir = os.path.abspath(basedir)
+
+AVOCADO = os.environ.get("UNITTEST_AVOCADO_CMD", "./scripts/avocado")
+
+TEST_SETUP_EXCEPTION = """
+import avocado
+
+class AvocadoTest(avocado.Test):
+
+    def setUp(self):
+        self.log.info('setup code before')
+        raise
```

Comment:
Also this is very similar to existing test defined in `functional/test_canceltests.py`, how about making this test generic, putting it directly in `selftests/.data/` and reusing it everywhere? Also this selftest does not cover all possible situation and if we are (and I think we should) to test all variants, we'd benefit from a generic test which raises/cancels on various places logging messages all around.

I imagine at least:

```yaml
location: !mux
    import:
    setUp:
    test:
    tearDown:
exception: !mux
    self.skip:
    self.cancel:
    self.fail:
    exception:
```

And this test can replace some of the `test_basic` ones as it checks all the statuses (if we include `self.log.warn(...)` which would also make sense). and some of the `skip` and `cancel` tests. Putting it all to one place would IMO make sense and there is no need to create the test file for every variant, simply use `exc = self.params.get("setUpFailure"); if exc: ...`.